sqlglot/dialects/oracle.py

```diff
@@ -282,3 +282,180 @@ def add_column_sql(self, expression: exp.AlterTable) -> str:
+    transformation is done in place.
+
+    Args:
+        node: The node to clean"""
+    if isinstance(node, exp.Binary):
+        if node.left is None:
+            node.replace(node.right)
+        elif node.right is None:
+            node.replace(node.left)
+
+
+def _has_join_mark(col: exp.Expression) -> bool:
+    """Check if the column has a join mark
+
+    Args:
+        The column to check
+    """
+    if not isinstance(col, exp.Column):
+        return False
+    result = col.args.get("join_mark", False)
+    return result
+
+
+def _equality_to_join(
+    eq: exp.Binary, old_joins: t.Dict[str, exp.Join], old_from: exp.From
+) -> t.Optional[exp.Join]:
+    """Convert an equality predicate to a join if it contains a join mark
+
+    Args:
+        eq: The equality expression to convert to a join
+
+    Returns:
+        The join expression if the equality contains a join mark (otherwise None)
+    """
+    if not (isinstance(eq.left, exp.Column) or isinstance(eq.right, exp.Column)):
+        return None
+    new_eq = eq.copy()
+    left_has_join_mark = _has_join_mark(eq.left)
+    right_has_join_mark = _has_join_mark(eq.right)
+
+    if left_has_join_mark:
+        new_eq.left.set("join_mark", False)
+        join_on = new_eq.left.table
+    elif right_has_join_mark:
+        new_eq.right.set("join_mark", False)
+        join_on = new_eq.right.table
+    else:
+        return None
+
+    join_this = old_joins.get(join_on, old_from).this
+    return exp.Join(this=join_this, on=new_eq, kind="LEFT")
+
+
+def _eliminate_join_marks_from_scope(scope) -> None:
+    """Remove join marks columns in scope's where clause.
+    Converts them to left joins and replaces any existing joins.
+    Updates scope in place.
+
+    Args:
+        scope: The scope to remove join marks from
+    """
+    select_scope = scope.expression
+    if not select_scope.args.get("where"):
+        return
+    if not select_scope.args.get("joins"):
+        return
+
+    # dictionaries used to keep track of joins to be replaced
+    old_joins: t.Dict[str, exp.Join] = {
+        join.alias_or_name: join for join in list(select_scope.args.get("joins", []))
+    }
+    new_joins: t.Dict[str, exp.Join] = {}
+
+    for node in scope.find_all(exp.Column):
+        if _has_join_mark(node):
+            # find the predicate that contains the join mark
+            predicate = node.find_ancestor(exp.Predicate)
+            if not isinstance(predicate, exp.Binary):
+                continue
+            predicate_parent = predicate.parent
+
+            join_on = predicate.pop()
+            new_join = _equality_to_join(
+                join_on, old_joins=old_joins, old_from=select_scope.args["from"]
+            )
+            if new_join:
+                new_joins = _update_join_dict(new_join, new_joins)
+            if predicate_parent:
+                _clean_binary_node(predicate_parent)
```

Comment:
Let's not create short helpers like this one when they're not used in multiple places. It makes understanding the code harder because you have to jump back and forth between use site and implementation site. I also found the verb "clean" to be somewhat confusing in this context.

Reply:
removed